src/util/amount.rs

```diff
@@ -68,22 +76,26 @@ impl FromStr for Denomination {
 
     /// Convert from a str to Denomination.
     ///
-    /// Any combination of upper and/or lower case, excluding uppercase 'M' is considered valid.
-    /// - Singular: BTC, mBTC, uBTC
+    /// Any combination of upper and/or lower case, excluding uppercase of SI(m, u, n, p) is considered valid.
+    /// - Singular: BTC, mBTC, uBTC, nBTC, pBTC
     /// - Plural or singular: sat, satoshi, bit, msat
     ///
-    /// Due to ambiguity between mega and milli we prohibit usage of leading capital 'M'.
+    /// Due to ambiguity between mega and milli, pico and peta we prohibit usage of leading capital 'M', 'P'.
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         use self::ParseAmountError::*;
+        use self::Denomination as D;
 
-        if s.starts_with('M') {
-            return Err(denomination_from_str(s).map_or_else(
-                || UnknownDenomination(s.to_owned()),
-                |_| PossiblyConfusingDenomination(s.to_owned())
-            ));
+        let starts_with_uppercase = || s.starts_with(|ch: char| ch.is_uppercase());
```

Comment:
Post-merge nit for a follow-up PR: `s.starts_with(char::is_uppercase);`

Reply:
Soon as edition 2018 lands we seriously have to get Clippy running on CI, this stuff is wasting precious reviewer resources :)